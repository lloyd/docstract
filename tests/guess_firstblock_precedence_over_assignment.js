// This test checks the precedence rules of guessing functions.
// specifically the first block should be considered the module
// documentation, even when there's assignment in the code chunk
// after it.

/**
 * Allows one to control fullscreen view for the main application window
 */

var mainWin = require("window-utils");

/**
 * Size the main application window to consume the full screen
 */
exports.enable = function() {
    mainWin.activeWindow.fullScreen=true;
};

/**
 * Disable fullscreen mode (noop if it wasn't enabled)
 */
exports.disable = function() {
    mainWin.activeWindow.fullScreen=false;
};

/**
 * Toggle fullscreen.
 */
exports.toggle = function() {
    mainWin.activeWindow.fullScreen=!mainWin.activeWindow.fullScreen;
};
