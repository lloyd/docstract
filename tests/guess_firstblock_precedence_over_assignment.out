{
  "desc": "Allows one to control fullscreen view for the main application window", 
  "filename": "tests/guess_firstblock_precedence_over_assignment.js", 
  "functions": [
    {
      "desc": "Size the main application window to consume the full screen", 
      "name": "enable", 
      "source_lines": [
        12, 
        14
      ]
    }, 
    {
      "desc": "Disable fullscreen mode (noop if it wasn't enabled)", 
      "name": "disable", 
      "source_lines": [
        19, 
        21
      ]
    }, 
    {
      "desc": "Toggle fullscreen.", 
      "name": "toggle", 
      "source_lines": [
        26, 
        28
      ]
    }
  ], 
  "module": "guess_firstblock_precedence_over_assignment", 
  "source_lines": [
    6, 
    9
  ]
}
